Xamarin.PropertyEditing.Mac/Controls/Custom/NumericSpinEditor.cs

```diff
@@ -0,0 +1,287 @@
+﻿using System;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	public class NumericSpinEditor : NSView
+	{
+		protected NumericTextField numericEditor;
+		protected NSStepper stepper;
+		protected bool editing;
+
+
+		public event EventHandler ValueChanged;
+		public event EventHandler StepperValueChanged;
+		public event EventHandler FieldEditingEnded;
+
+		public ValidationType NumericMode {
+			get { return numericEditor.NumericMode; }
+			set {
+				numericEditor.NumericMode = value;
+				Reset ();
+			}
+		}
+
+		public string StringValue {
+			get { return numericEditor.StringValue; }
+			set { numericEditor.StringValue = value; }
```

Comment:
@alanmcgovern the broken syncing is down to the fact that SetValue, isn't being used whereever the value is set/changed. Using that method appears to fix the syncing issue.